src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    public static final int DEFAULT_SIZE_DATA = 10;
+    public static final double RESIZE_INDEX = 1.5;
+    private int size;
+    private T[] dataObject;
+
+    public ArrayList() {
+        dataObject = (T[]) new Object[DEFAULT_SIZE_DATA];
+    }
+
     @Override
     public void add(T value) {
-
+        getArrayList();
+        dataObject[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
 
+        getIndexFrom(index);
+        getArrayList();
+        System.arraycopy(dataObject, index, dataObject, index + 1, size - index);
+        dataObject[index] = value;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        indexValidation(index);
+        return dataObject[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        indexValidation(index);
+        dataObject[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        indexValidation(index);
+        T deleteArray = dataObject[index];
+        size--;
+        System.arraycopy(dataObject, index + 1, dataObject, index, size - index);
+        return deleteArray;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        for (int j = 0; j < size; j++) {
+            if (element != null
+                    && element.equals(dataObject[j])
+                    || element == dataObject[j]) {
+                return remove(j);
+            }
+        }
+        throw new NoSuchElementException("Element not found" + element);
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private void indexValidation(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("Index is not supported" + index);
+        }
+    }
+
+    private void getIndexFrom(int index) {
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index is not supported" + index);
```

Comment:
why index is not supported? What is cause? 
don't forget about spaces during concatenation